api/jobs.go

```diff
@@ -938,7 +941,7 @@ func (j *Job) Canonicalize() {
 		j.Namespace = pointerOf(DefaultNamespace)
 	}
 	if j.Priority == nil {
-		j.Priority = pointerOf(50)
+		j.Priority = pointerOf(DefaultPriority)
```

Comment:
Sorry I missed this when I was reviewing the code, but this isn't quite going to work. When I checked this PR out locally and ran it, I discovered that the new `job_default_priority` isn't being respected. That's because when you submit a job without a `priority` set, it's defaulting to this hard-coded value instead of the value from the server.

I think what we're going to want to do here is set the value to 0 and then on the server treat `0` as "unset" so that it falls back to the default.

Reply:
Nice catch! All those mutations are a bit confusing :exploding_head:

I've pushed a new commit, but I'm not sure if it's enough. :thinking:  

Should we also update the `ValidateJobRequest`? (command/agent/job_endpoint.go)

```go
func (s *HTTPServer) ValidateJobRequest(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
...
	job := ApiJobToStructJob(validateRequest.Job)
	if job.Priority == 0 {
		job.Priority = s.agent.Server().GetConfig().JobDefaultPriority
	}
...
```